File: problems/3550-smallest-index-with-digit-sum-equal-to-index/solution.cpp
class Solution {
public:
    int smallestIndex(vector<int>& nums) {
        int n = nums.size();
        for (int i = 0; i < n; ++i) {
            int sum = 0;
            int temp = nums[i];
            while (temp > 0) {
                sum += temp % 10;
                temp /= 10;
            }
            if (sum == (i % 10)) {
                return i;
            }
        }
        return -1;
    }
};

File: problems/3550-smallest-index-with-digit-sum-equal-to-index/solution.py
class Solution:
    def smallestIndex(self, nums: List[int]) -> int:
        for i in range(len(nums)):
            s = 0
            num = nums[i]
            str_num = str(num)
            for digit in str_num:
                s += int(digit)
            if s == i:
                return i
        return -1